app/services/promo_registrar.rb

```diff
@@ -35,8 +39,14 @@ def register_channel(channel)
     if e.response[:status] == 409
       Rails.logger.warn("PromoRegistrar #register_channel returned 409, channel already registered.  Using PromoRegistrationGetter to get the referral_code.")
       Rails.logger.info("Attempted to register channel #{channel.details.channel_identifier}, but it already registered.  Saving referral code #{referral_code}.")
-      referral_code = PromoRegistrationGetter.new(publisher: @publisher, channel: channel).perform
-      referral_code
+      # Get the referral code
+      registration = PromoRegistrationGetter.new(publisher: @publisher, channel: channel).perform
+
+      if registration["owner_id"] != @publisher.id
```

Comment:
Nitpicking as this is a best practice: `register_channel(channel)`'s currently returning `referral_code` as part of the return type and `@should_update_promo_server` as an instance_variable.  This can be improved where the return type is a hash, that way we don't have code that passes a private variable as an instance variable. 

e.g. `{referral_code: referral_code, should_update_promo_server: false}`

Reply:
I think this is a good idea, will change.